src/support/thread_group.c

```diff
@@ -92,8 +93,27 @@ __thread_group_shrink(
 		 */
```

Comment:
This comment looks strange to me, preceding a bunch of code that has nothing to do with waking threads? (I realize it's in the original code, but figured I'd mention it.)